app/src/main/java/com/sun/heartrate/ui/saveheartbeat/SaveHeartbeatFragment.kt

```diff
@@ -0,0 +1,109 @@
+package com.sun.heartrate.ui.saveheartbeat
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import com.sun.heartrate.R
+import kotlinx.android.synthetic.main.fragment_save_heartbeat.*
+
+class SaveHeartbeatFragment(
+    private val onBackPressed: OnBackPressed
+) : Fragment(),
+    View.OnClickListener {
+    
+    var isStatusLast = GENERAL
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        retainInstance = true
+    }
+    
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? = inflater.inflate(R.layout.fragment_save_heartbeat, container, false)
+    
+    
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initView()
+    }
+    
+    private fun initView() {
+        assignViews(
+            buttonSave,
+            imageViewBack,
+            imageViewBeforeTraining,
+            imageViewBeforeTraining,
+            imageViewAfterTraining,
+            imageViewGeneral,
+            imageViewHeavyTraining,
+            imageViewRested
+        )
+    }
+    
+    private fun View.OnClickListener.assignViews(vararg views: View?) {
+        views.forEach { it?.setOnClickListener(this) }
+    }
+    
+    override fun onClick(view: View?) {
+        when (view?.id) {
+            R.id.imageViewBack -> onBackPressed.backFragment()
+            R.id.buttonSave -> {
```

Comment:
Viết 1 method mới xử lý việc save và back.
```suggestion
            R.id.buttonSave -> saveDataThenBack()
```